File: GiaPhaHub-be/GiaPhaHub.Domain/Entities/FamilyMember.cs
using GiaPhaHub_be.Domain.Abstractions.Base;

namespace GiaPhaHub_be.Domain.Entities;

public class FamilyMember : EntityTrackable<int>
{
    public string Name { get; set; } = string.Empty;

    public string Gender { get; set; } = string.Empty;

    public DateTime BirthDate { get; set; }

    public DateTime? DeathDate { get; set; }

    public string? Avatar { get; set; }

    public string? Phone { get; set; }

    public string? Address { get; set; }

    public string? Bio { get; set; }
    public int Generation { get; set; }

    // thuộc cây gia phả nào
    public int FamilyTreeId { get; set; }

    public FamilyTree FamilyTree { get; set; } = null!;

    // relationship graph
    public ICollection<Relationship> FromRelationships { get; set; } = new List<Relationship>();

    public ICollection<Relationship> ToRelationships { get; set; } = new List<Relationship>();
}

File: GiaPhaHub/src/models/FamilyMember.ts
import type { RelationshipResponse } from "@/models/Relationship";

export interface FamilyMemberResponse {
  id: number;
  name: string;
  gender: string;
  birthDate: string;
  deathDate: string;
  avatar: string | null;
  phone: string | null;
  address: string | null;
  bio: string | null;
  generation: number;
  fatherId: number | null;
  motherId: number | null;
  children: number[];
  spouses: number[];
  members: Record<number, FamilyMemberResponse> | FamilyMemberResponse[];
  fromRelationships?: RelationshipResponse[];
  toRelationships?: RelationshipResponse[];
}
export interface FamilyMemberRequest {
  name: string;
  gender: string;
  birthDate: string;
  deathDate: string;
  avatar: string;
  phone: string;
  address: string;
  bio: string;
  parentId: number;
  spouseId: number;
  spouseRelationship: string;
  generation: number;
}
